src/rotary-dialer-element.ts

```diff
@@ -0,0 +1,297 @@
+import { css, customElement, html, LitElement, property } from 'lit-element';
+
+type Digit = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9;
+enum Direction {
+  normal = 'normal',
+  reverse = 'reverse',
+}
+
+@customElement('wokwi-rotary-dialer')
+export class RotaryDialerElement extends LitElement {
+  @property() value: number | '' = 0;
+
+  static get styles() {
+    return css`
+      .text {
+        cursor: grab;
+        user-select: none;
+      }
+      input:focus + svg #container {
+        stroke: #4e50d7;
+        stroke-width: 3;
+      }
+      .hide-input {
+        position: absolute;
+        clip: rect(0 0 0 0);
+        width: 1px;
+        height: 1px;
+        margin: -1px;
+      }
+      .dialer-anim {
+        transform-origin: center;
+      }
+      @keyframes dialOne {
+        0% {
+          transform: rotate(0);
+        }
+        50% {
+          transform: rotate(56deg);
```

Comment:
Not sure if that would work, but perhaps you can store the target degree in a CSS variable and in this way reuse a single animation for all the digits?

Another way I'm thinking off would be to replace the animations with transitions, and then when a digit is dialed, you apply a class to start the transition on the `transform` property. When it ends you remove that class. Not sure if it'd be better than the current approach, just an idea...

Reply:
css variable won't work for the @keyframe AFAIK I could try insert a new rule (hardcore), regarding transitions if I'll use that than I could use css var for the transform - let me try that direction :) I like it

Reply:
@urish removing the transition class onanimationend didn't work :/ I'll DRY the anaimation code

Reply:
When working with CSS transitions, it seems like the event you want to listen for is [`transitionend`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event). Have you tried that?

Reply:
hey just noticed, no I didnt try that! I naively assumed that it uses the onanimationend, glad to learn something new, will give. it a try!